README.md

```diff
@@ -0,0 +1,73 @@
+# pydarn
+[![License: GPL v3](https://img.shields.io/badge/License-GPLv3-blue.svg)](https://www.gnu.org/licenses/gpl-3.0) [![Python 3.6](https://img.shields.io/badge/python-3.6-blue.svg)](https://www.python.org/downloads/release/python-360/) [![GitHub version](https://badge.fury.io/gh/boennemann%2Fbadges.svg)](http://badge.fury.io/gh/boennemann%2Fbadges)
+
+SuperDARN data visualization library. 
+
+## Getting Started 
+
+The following instructions will allow you to install and give some examples on how to use pydarn. 
+
+### Prerequisites
+
+**python 3.6+**
+
+| Ubuntu      | OpenSuse       | Fedora        |
+| ----------- | -------------- | ------------- |
+| libyaml-dev | python3-PyYAML | libyaml-devel |
+
+You can check your python version with  
+`$ python --version` or 
+`$ python3 --version`
+### Installing 
+
+1. Clone git repository:   
+   `git clone https://github.com/SuperDARN/pydarn.git`
+
+2. Installing pydarn  
+    1. **Recommended**: Installing a [virtual environment](https://packaging.python.org/guides/installing-using-pip-and-virtualenv/), this option allows the library to install needed version of libraries without affecting system libraries.  
+        * First install the environment:  
+      `$ python3 -m pip install --user virtualenv`  
+      `$ python3 -m virtualenv <environment name>`  
+		  `$ source <environment name>/bin/activate`
+        * Navigate to where you cloned pydarn:  
+		  `$ python3 setup.py install`
+    2. Install in the system (root privileges required):  
+		   `$ sudo python3 setup.py install`
+
+### Examples
+
+#### Reading DMAP file 
+The following example shows how to read in a FITACF file, one of the SuperDARN's DMAP file types. 
+
+```python
+import pydarn
+dmap_file = "./20180410.C0.sas.fitacf"
+dmap_reader = pydarn.DmapRead(dmap_file)
+dmap_data = dmap_reader.read_records() 
+
+# dmap_data[record number][paramter name].value
+print(dmap_data[0]['bmnum'].value) 
```

Comment:
Is it possible to create a getter or some sort of decorator so that the user doesn't have to explicitly call .value on every data value they want? ex. if i call dmap_data[0]['bm_num'] then under the hood it returns me dmap_data[0]['bm_num'].value by default? I don't think this affects too much, but if you are writing a large script or something you are going to have a lot of extraneous looking code for .value.

Reply:
Personally, I like the `.value` because it is explicit and shows self-documenting code (best practices). It also keeps things consistent. Especially with the complaints that RST and DaVitpy have gotten with unclear variable names and code.  

For example: 
```python
for i in range(record[0]['beam'].shape[1] ):
      amplify *= record[0]['beam'].value
```
vs. 
```python
for i in range(record[0]['beam'].shape[1] ):
      amplify *= record[0]['beam']    
```

The former is clearer and easier to understand at first glance (which is me in debugging mode) and the latter is more confusing or black magic.  

Or to avoid extraneous looking code you can just store it in a variable ahead of time and then use it later to make the lines smaller. 

For Example: 
```python
Plot (record[0][velocity'].value,  record[0]['time'].value, default .... plot ... settings)
```
vs.
```python 
velocity_data = record[0][velocity'].value
time_data = record[0]['time'].value
Plot (velocity_data,  time_data, default .... plot ... settings)
```
The latter is also considered cleaner code too. I mean at the end of the day you are going to have to access the `record[0][velocity']` to get most of your data (assuming no loops are involved to cut some stuff down) so an extra `.value` especially with autocomplete seems okay to me :) 

Also, I don't know if you can make a getter function/decorator with a collection containing a collection. You are potentially looking at some overhead with a getter method and that means a decrease in performance.

Reply:
Sounds good. I think this is partly a preference thing and I don't think there are necessarily any right or wrong answers here. I see a couple of thumbs ups on my suggestion, so maybe some others can comment on what they think since this is ultimately a decision on top level functionality. Personally, I still stick by my suggestion as a preference provided there is no performance hits.